packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.kt

```diff
@@ -342,6 +332,23 @@ class LocationModule : Module(), LifecycleEventListener, SensorEventListener, Ac
     } ?: throw NoPermissionsModuleException()
   }
 
+  private fun getProviderStatus(): LocationProviderStatus {
+    val manager = mContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+
+    val isGpsAvailable = manager.isProviderEnabled(LocationManager.GPS_PROVIDER)
+    val isNetworkAvailable = manager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
+    val isLocationServicesEnabled = manager.isProviderEnabled(LocationManager.GPS_PROVIDER) || manager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
```

Comment:
is checking providers enough to indicate whether location service is enabled? smart-location has the code at https://github.com/mrmans0n/smart-location-lib/blob/bfefb04770bb3a9653515a526c01908c36ed1c66/library/src/main/java/io/nlopez/smartlocation/location/utils/LocationState.java#L37-L54

maybe we can use [`LocationManagerCompat.isLocationEnabled`](https://developer.android.com/reference/androidx/core/location/LocationManagerCompat#isLocationEnabled(android.location.LocationManager))

Reply:
We don't support devices on versions 19 and below so that code isn't necessary. Good catch! We can use `LocationManagerCompat`. It does the same as I've done on apis 28 and below and I didn't see any issues running it on the latest android version but it's better to use it 👍